DeviceSimulation/SimulationTests.cs

```diff
@@ -1,33 +1,299 @@
-﻿using System.Net;
+﻿using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Helpers;
 using Helpers.Http;
+using Microsoft.Azure.Devices;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 using Xunit;
 
 namespace DeviceSimulation
 {
     public class SimulationTests
     {
         private readonly IHttpClient httpClient;
-        private const string DS_ADDRESS = "http://127.0.0.1:9003/v1";
-
+        private readonly RegistryManager registry;
+        private readonly string IOTHUB_CONNECTION_STRING;
         public SimulationTests()
         {
+            var config = new ConfigurationBuilder()
+                .AddJsonFile("settings.json")
+                .AddEnvironmentVariables()
+                .Build();
+            this.IOTHUB_CONNECTION_STRING = config["PCS_IOTHUB_CONNSTRING"];
             this.httpClient = new HttpClient();
         }
 
         /// <summary>
         /// Integration test using a real HTTP instance.
         /// Test that the service starts normally and returns ok status
         /// </summary>
-        [Fact]
+        [Fact, Trait("Type", "IntegrationTest")]
         public void Should_Return_OK_Status()
         {
             // Act
-            var request = new HttpRequest(DS_ADDRESS + "/status");
+            var request = new HttpRequest(Constants.DS_ADDRESS + "/status");
             request.AddHeader("X-Foo", "Bar");
             var response = this.httpClient.GetAsync(request).Result;
 
             // Assert
             Assert.Equal(HttpStatusCode.OK, response.StatusCode);
         }
+
+        [Fact, Trait("Type", "IntegrationTest")]
+        public void Should_Return_Currently_Running_Simulation()
+        {
+            //Arrange
+            this.Should_Delete_Existing_Simulation();
+            var simulation = JObject.Parse(@"{   
+                'Enabled': true, 
+                'DeviceModels': [   
+                    {   
+                        'Id': 'truck-01', 
+                        'Count': 5 
+                    } 
+                ]
+            }");
+            var createSimulationRequest = new HttpRequest(Constants.DS_ADDRESS + "/simulations");
+            createSimulationRequest.SetContent(simulation);
+            var createSimulationResponse = this.httpClient.PostAsync(createSimulationRequest).Result;
+            Assert.Equal(HttpStatusCode.OK, createSimulationResponse.StatusCode);
+
+            //Act
+            var getCurrentSimulationRequest = new HttpRequest(Constants.DS_ADDRESS + "/simulations/1");
+            var getCurrentSimulationResponse = this.httpClient.GetAsync(getCurrentSimulationRequest).Result;
+
+            //Assert
+            Assert.Equal(HttpStatusCode.OK, getCurrentSimulationResponse.StatusCode);
```

Comment:
Doesn't this check just confirm that you're receiving an HTTP 200 response code? If the spirit of the test is to verify that we're returned the currently running simulation, we should check the body to confirm that the simulation returned matches what we expect, and not just that we received a 200.